network/src/protocols/rpc/mod.rs

```diff
@@ -288,29 +412,44 @@ where
                     debug!("Rpc client canceled outbound rpc call to {}", peer_id_str);
                 },
             }
-        }
+            // Return the request_id for state management in the main event-loop.
+            request_id
+        };
+        outbound_rpc_tasks.push(f.boxed());
     }
 }
 
-async fn handle_outbound_rpc_inner<TSubstream>(
-    mut peer_handle: PeerHandle<TSubstream>,
+async fn handle_outbound_rpc_inner(
+    mut peer_handle: PeerHandle,
+    request_id: RequestId,
     protocol: ProtocolId,
     req_data: Bytes,
-) -> Result<Bytes, RpcError>
-where
-    TSubstream: AsyncRead + AsyncWrite + Send + Unpin,
-{
-    let _timer = counters::LIBRA_NETWORK_RPC_LATENCY.start_timer();
-    // Request a new substream with the peer.
-    let substream = peer_handle.open_substream(protocol).await?;
-    // Rpc messages are length-prefixed.
-    let mut substream = Framed::new(IoCompat::new(substream), LengthDelimitedCodec::new());
-    // Send the rpc request data.
+    response_rx: oneshot::Receiver<RpcResponse>,
+) -> Result<Bytes, RpcError> {
     let req_len = req_data.len();
-    substream.buffered_send(req_data).await?;
-    // We won't send anything else on this substream, so we can half-close our
-    // output side.
-    substream.close().await?;
+    let peer_id = peer_handle.peer_id();
+
+    // Create NetworkMessage to be sent over the wire.
+    let request = NetworkMessage::RpcRequest(RpcRequest {
+        request_id,
+        // TODO: Use default priority of 0 for now. To be exposed via network API.
+        priority: 0,
```

Comment:
maybe make a const somewhere called default priority that way you don't need this todo?

Reply:
The TODO is mainly to suggest that it is our intention to expose setting priority as part of Network API instead of always using the default value